experimental/sso-with-skills/SkillBot/SkillAdapterWithErrorHandler.cs

```diff
@@ -0,0 +1,104 @@
+            : base(configuration, credentialProvider, authConfig, logger: logger)
+        {
+            _conversationState = conversationState;
+            _userState = userState;
+
+            OnTurnError = async (turnContext, exception) =>
+            {
+                // Log any leaked exception from the application.
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                var errorMessageText = "The skill encountered an error or bug.";
+                var errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.IgnoringInput);
+                await turnContext.SendActivityAsync(errorMessage);
+
+                errorMessageText = "To continue to run this bot, please fix the bot source code.";
+                errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.ExpectingInput);
+                await turnContext.SendActivityAsync(errorMessage);
+
+                // Send a trace activity, which will be displayed in the Bot Framework Emulator
+                // Note: we return the entire exception in the value property to help the developer, this should not be done in prod.
+                await turnContext.TraceActivityAsync("OnTurnError Trace", exception.ToString(), "https://www.botframework.com/schemas/error", "TurnError");
+
+                // Send and EndOfConversation activity to the skill caller with the error to end the conversation
+                // and let the caller decide what to do.
+                var endOfConversation = Activity.CreateEndOfConversationActivity();
+                endOfConversation.Code = "SkillError";
+                endOfConversation.Text = exception.Message;
+                await turnContext.SendActivityAsync(endOfConversation);
+
+                try
+                {
+                    // Delete the conversationState for the current conversation to prevent the
+                    // bot from getting stuck in a error-loop caused by being in a bad state.
+                    // ConversationState should be thought of as similar to "cookie-state" in a Web pages.
+                    await _conversationState.DeleteAsync(turnContext);
+                }
+                catch (Exception ex)
+                {
+                    logger.LogError(ex, $"Exception caught on attempting to Delete ConversationState : {ex}");
+                }
+            };
+        }
+
+        public override async Task<InvokeResponse> ProcessActivityAsync(ClaimsIdentity claimsIdentity, Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            if (!claimsIdentity.Claims.Any(x => x.Type == "azp"))
```

Comment:
We want to remove this workaround as this is now natively supported by the SDK and Token service